discovery-provider/src/api/v1/users.py

```diff
@@ -837,26 +852,41 @@ def get(self, id):
 
 @full_ns.route("/content_node/<string:replica_type>", doc=False)
 class UsersByContentNode(Resource):
-    @full_ns.marshal_with(users_by_content_node_response)
-    # @cache(ttl_sec=30)
-    def get(self, replica_type):
-        """New route to call get_users_cnode with replica_type param (only consumed by content node)
-        - Leaving `/users/creator_node` above untouched for backwards-compatibility
-
+    @ns.doc(
+        id="""Get Users By Replica Type for Content Node""",
+        description="""
+        (Only consumed by Content Node) Gets users that have a given Content Node endpoint as
+        their primary, secondary, or either (depending on the replica_type passed).
         Response = array of objects of schema {
             user_id, wallet, primary, secondary1, secondary2, primarySpId, secondary1SpID, secondary2SpID
         }
-        """
+        """,
+        responses={200: "Success", 400: "Bad request", 500: "Server error"},
+    )
+    @full_ns.marshal_with(users_by_content_node_response)
+    @cache(ttl_sec=30)
+    def get(self, replica_type):
         args = users_by_content_node_route_parser.parse_args()
 
-        cnode_url = args.get("creator_node_endpoint")
+        # Endpoint that a user's primary/secondary/either must be set to for them to be included in the results
+        cnode_url = args.get("content_node_endpoint") or args.get(
+            "creator_node_endpoint"
+        )
+        # Offset after sorting user_id in ascending order (using > comparison in SQL query)
```

Comment:
can we find and link to a doc that explains why this pagination approach is used here? its a new pattern so would be good to document for others' reference

also lets def share out the explanation behind this approach with the team 😎

Reply:
absolutely. just changed that and made a post about it in #eng